src/shillelagh/backends/apsw/vt.py

```diff
@@ -389,11 +406,47 @@ def BestIndex(  # pylint: disable=too-many-locals
             estimated_cost,
         )
 
+    def BestIndexObject(self, index_info: apsw.IndexInfo) -> bool:
+        """
+        Alternative to ``BestIndex`` that allows returning only selected columns.
+        """
+        columns = self.adapter.get_columns()
+        column_names = list(columns.keys())
+        self.requested_columns = {column_names[i] for i in index_info.colUsed}
```

Comment:
is there no way to do this without binding `requested_columns` to self? Seem like that might be a foot gun down the road. Perhaps encoding the list in `index_name` ?

Reply:
I think that would actually clean the code up a fair bit.